api/v1alpha1/virtualmachine_types.go

```diff
@@ -47,6 +47,18 @@ type VirtualMachinePort struct {
 	Protocol corev1.Protocol `json:"protocol"`
 }
 
+// NetworkInterfaceProviderReference contains info to locate a network interface provider object.
+type NetworkInterfaceProviderReference struct {
+	// APIGroup is the group for the resource being referenced.
+	APIGroup string `json:"apiGroup"`
+	// Kind is the type of resource being referenced
+	Kind string `json:"kind"`
+	// Name is the name of resource being referenced
+	Name string `json:"name"`
+	// API version of the referent.
+	APIVersion string `json:"apiVersion,omitempty"`
```

Comment:
Why omitempty just for this?

Reply:
Good question. I was using a pattern similar to our netop types here. 

But seems like the use case would be when we have a fallback version later down the line. v1a1 or v1.